### ISSUE TITLE
UI: Scroll bars missing in (deep) tracability view

### ISSUE THREAD

**Issue by haxtibal** [strictdoc-project/strictdoc#1938]
We have a large multi *.sdoc project. When going to traceability view there are no scrollbars, so we can view only a small portion of the tree.

- In traceability view there's a vertical scrollbar, but no horizontal scrollbar.
- In deep traceability view, there's neither vertical nor horizontal scrollbar.

Reproducible in Firefox and Chrome with latest strictdoc main branch revision.

PS: We've just started to explore the traceability views, first impression is this is a very useful feature. Thanks!

@mettta Guess this is mostly for you. If you can't reproduce, let me know what kind of debug information you would need. I'll also try to debug myself, but have a lot to learn in that area, will take time.

**Comment by stanislaw** [comment]
Hi @haxtibal , let's consider TR and DTR separately because their implementations are slightly different:

1) **TR:** Both @mettta and I are use macOS and we do have both the horizontal and vertical scrolling working (attaching my screen screenshot). Maybe we could add scrolling with keyboard keys additionally.

2) **DTR:** This screen doesn't have visible scrollbars but we can still scroll it to both vertically and horizontally, with shift + mouse wheel. The main behavior that we implemented there is panning with a space key: try dragging the screen with the space key pressed. Previously, it was possible to navigate down/up and right/left with just pressing arrow keys but that behavior was lost when switched to the new Zoom + Pan implementation. Here, we should probably display a UI view with a legend for navigation.

<img width="414" alt="Bildschirmfoto 2024-08-23 um 16 13 27" src="https://github.com/user-attachments/assets/3d81e542-789d-4b28-ae2a-500de863fc2b">

**Comment by haxtibal** [comment]
> Hi @haxtibal , let's consider TR and DTR separately because their implementations are slightly different:
> 
>     1. **TR:** Both @mettta and I are use macOS and we do have both the horizontal and vertical scrolling working (attaching my screen screenshot). Maybe we could add scrolling with keyboard keys additionally.

You're right. The horizontal scroll bar appears as soon as I shrink my browser windows. I wasn't aware that the TR view is intentionally truncated to the immediate parent and the immediate child. So the graph is long in vertical direction, but narrow in horizontal direction.

>     2. **DTR:** This screen doesn't have visible scrollbars but we can still scroll it to both vertically and horizontally, with shift + mouse wheel.

Works also for me. Wasn't aware of the Shift+wheel combination.

> The main behavior that we implemented there is panning with a space key: try dragging the screen with the space key pressed.

This is actually great :rocket: Wasn't aware of the combination.

> Previously, it was possible to navigate down/up and right/left with just pressing arrow keys but that behavior was lost when switched to the new Zoom + Pan implementation. Here, we should probably display a UI view with a legend for navigation.

Navigation is very good as it is when knowing the hotkeys. We can close this issue as case of PEBCAK. Thank you for explaining.

Idea for minor improvement (but maybe applies only in my specific case): When entering the DTR my screen is empty. That's because the viewport is at the upper left of the window and my graph has no ancestors for the first elements. But later elements have many ancestors. If the DTR started with a horizontally centered viewport, it would be assured that the initial viewport contains the document starting node.
